From 622cf77916a02fd44d821de95fe94ab9daf626c6 Mon Sep 17 00:00:00 2001
From: zhu youhua <youhua@codeaurora.org>
Date: Tue, 8 Dec 2015 17:57:22 +0800
Subject: [PATCH 8/9] Define EF MSPL/MLPL/PRL values and paths

Add several EF paths for UI to get CDMA device info.

Change-Id: Iff43d6b5a6706e54aaf50867419c30a1a322cb2f
CRs-Fixed: 2177396
---
 .../android/internal/telephony/uicc/CsimFileHandler.java    | 5 +++++
 .../com/android/internal/telephony/uicc/IccConstants.java   | 6 ++++++
 .../android/internal/telephony/uicc/RuimFileHandler.java    | 5 +++++
 3 files changed, 16 insertions(+)

diff --git a/src/java/com/android/internal/telephony/uicc/CsimFileHandler.java b/src/java/com/android/internal/telephony/uicc/CsimFileHandler.java
index c5dd05c0d..3a7b94802 100644
--- a/src/java/com/android/internal/telephony/uicc/CsimFileHandler.java
+++ b/src/java/com/android/internal/telephony/uicc/CsimFileHandler.java
@@ -43,8 +43,13 @@ public final class CsimFileHandler extends IccFileHandler implements IccConstant
         case EF_CSIM_IMSIM:
         case EF_CSIM_CDMAHOME:
         case EF_CSIM_EPRL:
+        case EF_CSIM_PRL:
         case EF_CSIM_MIPUPP:
+        case EF_RUIM_ID:
             return MF_SIM + DF_ADF;
+        case EF_CSIM_MSPL:
+        case EF_CSIM_MLPL:
+            return MF_SIM + DF_TELECOM + DF_MMSS;
         }
         String path = getCommonIccEFPath(efid);
         if (path == null) {
diff --git a/src/java/com/android/internal/telephony/uicc/IccConstants.java b/src/java/com/android/internal/telephony/uicc/IccConstants.java
index 0f41f1ef1..3479412a6 100644
--- a/src/java/com/android/internal/telephony/uicc/IccConstants.java
+++ b/src/java/com/android/internal/telephony/uicc/IccConstants.java
@@ -62,6 +62,7 @@ public interface IccConstants {
     // CDMA RUIM file ids from 3GPP2 C.S0023-0
     static final int EF_CST = 0x6F32;
     static final int EF_RUIM_SPN =0x6F41;
+    static final int EF_RUIM_ID = 0x6F31;
 
     // ETSI TS.102.221
     static final int EF_PL = 0x2F05;
@@ -72,6 +73,10 @@ public interface IccConstants {
     static final int EF_CSIM_IMSIM = 0x6F22;
     static final int EF_CSIM_CDMAHOME = 0x6F28;
     static final int EF_CSIM_EPRL = 0x6F5A;
+    static final int EF_CSIM_PRL = 0x6F30;
+    // C.S0074-Av1.0 Section 4
+    static final int EF_CSIM_MLPL = 0x4F20;
+    static final int EF_CSIM_MSPL = 0x4F21;
     static final int EF_CSIM_MIPUPP = 0x6F4D;
 
     //ISIM access
@@ -103,6 +108,7 @@ public interface IccConstants {
     static final String DF_GRAPHICS = "5F50";
     static final String DF_GSM = "7F20";
     static final String DF_CDMA = "7F25";
+    static final String DF_MMSS = "5F3C";
 
     //UICC access
     static final String DF_ADF = "7FFF";
diff --git a/src/java/com/android/internal/telephony/uicc/RuimFileHandler.java b/src/java/com/android/internal/telephony/uicc/RuimFileHandler.java
index 2323b5a4b..7f596b695 100644
--- a/src/java/com/android/internal/telephony/uicc/RuimFileHandler.java
+++ b/src/java/com/android/internal/telephony/uicc/RuimFileHandler.java
@@ -64,8 +64,13 @@ public final class RuimFileHandler extends IccFileHandler {
         case EF_CSIM_IMSIM:
         case EF_CSIM_CDMAHOME:
         case EF_CSIM_EPRL:
+        case EF_CSIM_PRL:
         case EF_CSIM_MIPUPP:
+        case EF_RUIM_ID:
             return MF_SIM + DF_CDMA;
+        case EF_CSIM_MSPL:
+        case EF_CSIM_MLPL:
+            return MF_SIM + DF_TELECOM + DF_MMSS;
         }
         return getCommonIccEFPath(efid);
     }
-- 
2.28.0

